mava/systems/jax/system.py

```diff
@@ -0,0 +1,139 @@
+# python3
+# Copyright 2021 InstaDeep Ltd. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Jax-based Mava system implementation."""
+import abc
+from types import SimpleNamespace
+from typing import Any, Callable, List
+
+from mava.core_jax import BaseSystem
```

Comment:
comment is more linked to one of your previous PR's but what is the interplay between the SystemBuilder class and the builder class? The systemBuilder is similar to our mava system class and the builder class allows one to add components as required.

Reply:
The `SystemBuilder` is the abstract builder class for the general builder that will be used across systems. The only abstract method the general builder will have is the `design` method which will be used by different systems to define the list of system components. The builder in this PR is just barebones for the testing. A more general system builder will be coming soon.